example/src/main/java/io/netty/example/http2/helloworld/server/Http2OrHttpHandler.java

```diff
@@ -14,45 +14,37 @@
  */
 package io.netty.example.http2.helloworld.server;
 
-import io.netty.channel.ChannelHandler;
-import io.netty.handler.codec.http2.Http2ConnectionHandler;
-import io.netty.handler.codec.http2.Http2OrHttpChooser;
-
-import javax.net.ssl.SSLEngine;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.ssl.ApplicationProtocolNegotiationHandler;
 
 /**
  * Negotiates with the browser if HTTP2 or HTTP is going to be used. Once decided, the Netty
  * pipeline is setup with the correct handlers for the selected protocol.
  */
-public class Http2OrHttpHandler extends Http2OrHttpChooser {
-    private static final int MAX_CONTENT_LENGTH = 1024 * 100;
+public class Http2OrHttpHandler extends ApplicationProtocolNegotiationHandler {
```

Comment:
So, it's gonna be about having a `Map<protocol_name, pipeline_builder>` in an `ApplicationProtocolNegotiationHandler` (APNH).  The problem is that APNH is a non-sharable handler, so we're gonna have to have as many `Map`s as the number of connections unless we share it.  Maybe the effort taken to build and keep such a map is not a win against just extending a method and do `switch-case` (which is allowed in JDK 8)? WDYT?